aggregator_core/src/lib.rs

```diff
@@ -70,15 +70,18 @@ impl<H: Handler> InstrumentedHandler<H> {
 
     async fn before_send(&self, conn: Conn) -> Conn {
         if let Some(span) = conn.state::<InstrumentedHandlerSpan>() {
+            let conn = self.0.before_send(conn).instrument(span.0).await;
```

Comment:
Apologies, this needs a clone
```suggestion
            let conn = self.0.before_send(conn).instrument(span.0.clone()).await;
```